File: prid1920-g13/ClientApp/src/app/models/Game.ts

export class Game {

    public id: number;
    public name: string;
    public deck: string;
    public expected_release_day: number;
    public expected_release_month: number;
    public expected_release_year: number;
    public platforms: string;
    public image: string;
    public year: string;

    constructor(data: any){
        if(data){
            console.log(data);
            this.id = data.id,
            this.name = data.name,
            this.deck = data.deck,
            this.platforms = data.platforms,
            this.expected_release_day = data.expected_release_day ? data.expected_release_day : 0,
            this.expected_release_month = data.expected_release_month ? data.expected_release_month : 0,
            this.expected_release_year = data.expected_release_year ? data.expected_release_year : 0,
            this.image = data.image ? data.image : 'uploads/_Unknown.png';

        }
    }

}

File: prid1920-g13/Models/ModelsEntity/Game.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace prid_1819_g13.Models
{
    public class Game
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Deck { get; set; }
        public int Expected_release_day {get;set;}
        public int Expected_release_month {get;set;}
        public int Expected_release_year {get;set;}
        public string Image { get; set; }
        public string Platforms { get; set; }
        public virtual ICollection<Event> Events {get;set;}
        [NotMapped]
        public virtual IList<UserGames> UserGames { get; set; } = new List<UserGames>();
        [NotMapped]
        public IEnumerable<User> Users
        {
            get => UserGames.Select(g => g.UserGame);
        }
    }


}
